packages/api/src/hooks/useActiveTyping.ts

```diff
@@ -36,15 +36,16 @@ function useActiveTyping(expireAfter?: number): readonly [Readonly<Record<string
   ]);
 
   const earliestExpireAt = Math.min(...Object.values(activeTypingState[0]).map(({ expireAt }) => expireAt));
-  const timeToRender = earliestExpireAt && earliestExpireAt - now;
 
   useEffect(() => {
+    const timeToRender = earliestExpireAt && earliestExpireAt - Date.now();
+
     if (timeToRender && isFinite(timeToRender)) {
       const timeout = setTimeout(forceRender, Math.max(0, timeToRender));
 
       return () => clearTimeout(timeout);
     }
-  }, [clearTimeout, forceRender, setTimeout, timeToRender]);
+  }, [clearTimeout, Date, earliestExpireAt, forceRender, setTimeout]);
 
```

Comment:
This change probably causes an additional re-render, but looking good to me.

Reply:
It's actually the otherwise. I may be wrong somewhere, so let me explain how it happened.

Before the change in this file, render was added/removed. Then, it caused the `timeToRender` to be inaccurate by 1 ms.

Says, t=900, `earliestExpireAt` is 1000, which means it should wake up at t=1000. `timeToRender` is computed to be 100ms. So, everything works perfectly.

But after some changes somewhere else, `earliestExpireAt` is still 1000. But it woke up at t=999 instead of t=1000 (either now or `timeToRender` is wrong). It woke up 1 ms too early and t=999 didn't reach `expireAt`. So the logic did not expire some typing indicators.

Instead of passing `timeToRender`, which is a relative number. I am passing `earliestExpireAt` which is an absolute number. And it would calculate `timeToRender` inside the `useEffect`, so the timeout value should be more accurate.

Also, I think passing an absolute number would reduce the chance that this `useEffect` got re-render. `earliestExpireAt` is based on solely of all typing entries. `timeToRender` is based on all typing entries, *plus* `Date.now()`. So, `earliestExpireAt` should have a lesser chance to cause re-render.